test/functional/test_framework/test_framework.py

```diff
@@ -837,9 +838,16 @@ def _start_logging(self):
         # User can provide log level as a number or string (eg DEBUG). loglevel was caught as a string, so try to convert it to an int
         ll = int(self.options.loglevel) if self.options.loglevel.isdigit() else self.options.loglevel.upper()
         ch.setLevel(ll)
+
         # Format logs the same as bitcoind's debug.log with microprecision (so log files can be concatenated and sorted)
-        formatter = logging.Formatter(fmt='%(asctime)s.%(msecs)03d000Z %(name)s (%(levelname)s): %(message)s', datefmt='%Y-%m-%dT%H:%M:%S')
-        formatter.converter = time.gmtime
+        class MicrosecondFormatter(logging.Formatter):
+            def formatTime(self, record, _=None):
```

Comment:
by the way, I switched to `_=None` because `datefmt=None` runs into a linter error, and omitting the arg is not allowed (TypeErrors). Not sure if the linter is correct to complain about this.

Reply:
I don't have a strong opinion either, I just wanted to mention this because `_=None` may look weird to some - if the current approach is fine I'll just leave it at this.